lib/protobuf/wire.ex

```diff
@@ -53,44 +53,77 @@ defmodule Protobuf.Wire do
   def wire_type(:float), do: wire_32bits()
   def wire_type(mod) when is_atom(mod), do: wire_delimited()
 
+  @spec encode_from_wire_type(Protobuf.Wire.Types.wire_type(), term()) :: iodata()
+  def encode_from_wire_type(wire_type, value)
```

Comment:
I like the refactor, makes it more clear what's happening. Should it be inlined with `@compile {:inline, :encode_from_wire_type, 2}` to ensure the refactor does not affect performance? Not sure how performance sensitive this code is.

Reply:
Yes, that's a good call! Done.

Reply:
I was about to comment something along these lines too. Even if we inline this function, we'll still check against the wire type at run time, right? Since unknown fields should be the exception rather than the rule, I'm a bit on the fence about compromising performance on the general case. But it's hard to talk about it without running benchmarks, the impact might be negligible in the end.